### ISSUE TITLE
C1901 leads to false positives with numpy (as well)

### ISSUE THREAD

**Issue by mmarras** [astral-sh/ruff#4115]
Related to #3517: C1901 leads to false positives with numpy as well. 
```pyhton
desc = np.array(("","",""))
np.all(desc == "")  # is valid
np.all(not desc) # is not valid
```

`ruff 0.0263`

> PLC1901 `desc == ""` can be simplified to `not desc` as an empty string is falsey

The statment "can be simplified to `not desc`" is wrong.

**Comment by charliermarsh** [comment]
We made this rule require explicit opt-in, so closing for now.
